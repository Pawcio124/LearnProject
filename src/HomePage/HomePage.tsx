import { Box } from "@material-ui/core";

const HomePage = () => {
  return (
    <Box
      display={"flex"}
      height={"100%"}
      justifyContent={"center"}
      alignItems={"center"}
    >
      <h1>Welcome</h1>
    </Box>
  );
};

export default HomePage;
