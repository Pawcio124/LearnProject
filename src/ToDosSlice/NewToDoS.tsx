import { Box, Button, TextField } from "@material-ui/core";
import { useState } from "react";
import { useDispatch } from "react-redux";
import { addNewToDo } from "../app/store/toDoSlice";

const NewToDoS = () => {
  const [value, setValue] = useState<string>("");
  const dispatch = useDispatch();

  return (
    <Box>
      <TextField
        onChange={(e) => {
          setValue(e.target.value);
        }}
        variant="outlined"
        value={value}
      />
      <Button
        onClick={() => {
          dispatch(addNewToDo(value));
          setValue("");
        }}
      >
        Add
      </Button>
    </Box>
  );
};

export default NewToDoS;
