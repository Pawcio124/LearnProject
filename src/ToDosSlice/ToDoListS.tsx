import { Box, Button } from "@material-ui/core";
import { useDispatch, useSelector } from "react-redux";
import { RootState } from "../app/store/store";
import { removeToDo } from "../app/store/toDoSlice";

const ToDoListS = () => {
  const toDoList = useSelector((state: RootState) => state.toDos.toDos);
  const dispatch = useDispatch();

  return (
    <Box>
      {toDoList?.map((item, id) => (
        <div key={`item-${id}`}>
          {item}
          <Button
            onClick={() => {
              dispatch(removeToDo(id));
            }}
          >
            Delete
          </Button>
        </div>
      ))}
    </Box>
  );
};

export default ToDoListS;
