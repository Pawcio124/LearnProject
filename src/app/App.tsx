import { Routes, Route, Navigate, Link } from "react-router-dom";
import { AppBar, Box, Button, Toolbar } from "@material-ui/core";
import HomePage from "../HomePage/HomePage";
import ToDos from "../ToDos/ToDos";

function App() {
  return (
    <Box height={"100%"} display={"flex"} flexDirection={"column"}>
      <AppBar position="static">
        <Toolbar>
          <Button>
            <Link to={"/"}>Home</Link>
          </Button>
          <Button>
            <Link to={"/toDos"}>To Dos</Link>
          </Button>
        </Toolbar>
      </AppBar>
      <Box padding={2} flex={1} overflow={"auto"}>
        <Routes>
          <Route path="/" element={<HomePage />} />
          <Route path="/toDos" element={<ToDos />} />
          <Route path="*" element={<Navigate to="/" />} />
        </Routes>
      </Box>
      <Box
        display={"flex"}
        justifyContent={"center"}
        bgcolor={"gray"}
        height={"50px"}
        alignItems={"center"}
      >
        Footer
      </Box>
    </Box>
  );
}

export default App;
