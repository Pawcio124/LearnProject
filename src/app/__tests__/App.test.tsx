import { mount, shallow } from "enzyme";
import App from "../App";
import toJSON from "enzyme-to-json";
import { Provider } from "react-redux";
import { store } from "../store/store";
import { MemoryRouter } from "react-router-dom";
import HomePage from "../../HomePage/HomePage";
import { Button, TextField } from "@material-ui/core";
import ToDoListS from "../../ToDosSlice/ToDoListS";
import NewToDoS from "../../ToDosSlice/NewToDoS";

describe("App", () => {
  it("should render correctly", () => {
    const wrapper = shallow(<App />);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  it("should render Welcome", () => {
    const wrapper = mount(
      <MemoryRouter>
        <Provider store={store}>
          <App />
        </Provider>
      </MemoryRouter>
    );
    expect(wrapper.find(HomePage).text()).toEqual("Welcome");
  });

  it("should render toDo view", () => {
    const wrapper = mount(
      <MemoryRouter initialEntries={["/toDos"]}>
        <Provider store={store}>
          <App />
        </Provider>
      </MemoryRouter>
    );
    expect(wrapper.find(ToDoListS).find(Button).length).toEqual(0);
    wrapper
      .find(TextField)
      .find("input")
      .simulate("change", { target: { value: "test" } });
    wrapper.find(NewToDoS).find(Button).simulate("click");
    expect(wrapper.find(ToDoListS).find(Button).length).toEqual(1);
    wrapper.find(ToDoListS).find(Button).simulate("click");
    expect(wrapper.find(ToDoListS).find(Button).length).toEqual(0);
  });
});
