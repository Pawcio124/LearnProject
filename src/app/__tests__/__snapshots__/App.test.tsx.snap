// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render correctly 1`] = `
<Styled(MuiBox)
  display="flex"
  flexDirection="column"
  height="100%"
>
  <WithStyles(ForwardRef(AppBar))
    position="static"
  >
    <WithStyles(ForwardRef(Toolbar))>
      <WithStyles(ForwardRef(Button))>
        <Link
          to="/"
        >
          Home
        </Link>
      </WithStyles(ForwardRef(Button))>
      <WithStyles(ForwardRef(Button))>
        <Link
          to="/toDos"
        >
          To Dos
        </Link>
      </WithStyles(ForwardRef(Button))>
    </WithStyles(ForwardRef(Toolbar))>
  </WithStyles(ForwardRef(AppBar))>
  <Styled(MuiBox)
    flex={1}
    overflow="auto"
    padding={2}
  >
    <Routes>
      <Route
        element={<HomePage />}
        path="/"
      />
      <Route
        element={<ToDosS />}
        path="/toDos"
      />
      <Route
        element={
          <Navigate
            to="/"
          />
        }
        path="*"
      />
    </Routes>
  </Styled(MuiBox)>
  <Styled(MuiBox)
    alignItems="center"
    bgcolor="gray"
    display="flex"
    height="50px"
    justifyContent="center"
  >
    Footer
  </Styled(MuiBox)>
</Styled(MuiBox)>
`;
